account_move_cutoff/models/account_move.py

```diff
@@ -0,0 +1,217 @@
+            )
+            deferred_title = _("Advance revenue adjustment of %s (%s)") % (
+                self.name,
+                self.date.strftime("%m %Y"),
+            )
+        elif self.journal_id.type == "purchase":
+            cutoff_title = _("Advance expense recognition of %s (%s)") % (
+                self.name,
+                self.date.strftime("%m %Y"),
+            )
+            deferred_title = _("Advance expense adjustment of %s (%s)") % (
+                self.name,
+                self.date.strftime("%m %Y"),
+            )
+        return cutoff_title, deferred_title
+
+    def _prepare_deferred_entry(self, journal, date_, reference):
+        return {
+            "currency_id": journal.currency_id.id or journal.company_id.currency_id.id,
+            "move_type": "entry",
+            "line_ids": [],
+            "ref": reference,
+            "date": date_,
+            "journal_id": journal.id,
+            "partner_id": self.partner_id.id,
+            "cutoff_from_id": self.id,
+        }
+
+    def _create_cutoff_entries(self, lines_to_deferred):
+        # create one entry to move to the deferred_accrual_account_id
+        self.ensure_one()
+
+        entries = self.env["account.move"]
+        journal = self._get_deferred_journal()
+        # if not journal returns means no deferred entries to
+        # generate, we don't want any raises here
+        if not journal:
+            return
+
+        cutoff_title, deferred_title = self._get_deferred_titles()
+        periods = self._get_deferred_periods(lines_to_deferred)
+        cutoff_entry = self.create(
+            self._prepare_deferred_entry(
+                journal, fields.Date.to_string(self.date), cutoff_title
+            )
+        )
+        entries |= cutoff_entry
+
+        amounts = self._get_amounts_by_period(lines_to_deferred, periods)
+        # manage cutoff entry
+        for line, amounts_by_period in amounts:
+            line._create_cutoff_entry_lines(
+                cutoff_entry,
+                periods[0],
+                sum([amounts_by_period.get(p, 0) for p in periods[1:]]),
+            )
+
+        # manage deferred entries
+        for period in periods[1:]:
+            deferred_entry = self.create(
+                self._prepare_deferred_entry(
+                    journal,
+                    fields.Date.to_string(self._get_deferred_date_from_period(period)),
+                    deferred_title,
+                )
+            )
+            for line, amounts_by_period in amounts:
+                if self.currency_id.is_zero(amounts_by_period.get(period, 0)):
+                    continue
+
+                line._create_deferred_entry_lines(
+                    deferred_entry, period, amounts_by_period[period]
+                )
+
+            if deferred_entry.line_ids:
+                entries |= deferred_entry
+            else:
+                # TODO: not sure if the case is possible
```

Comment:
doesn't seems like a TODO since it's done, maybe remove `TODO` and explains this is a failsafe ?